### PR TITLE
Jumbotron componentclass

### PR DESCRIPTION
**Pull request by lo1tuma**
fixes: #878

### CODE REVIEW

File: src/Jumbotron.js [hunk 1]
```diff
@@ -2,11 +2,21 @@ import React from 'react';
 import classNames from 'classnames';
 
 const Jumbotron = React.createClass({
+  propTypes: {
+    componentClass: React.PropTypes.node.isRequired
```
Comment (jquense) [inline src/Jumbotron.js]: I think `node` might be the wrong type here...we are really looking for a string, or a function. i.e. tagname or Component Class.

File: test/NavbarSpec.js [hunk 2]
```diff
@@ -61,7 +61,7 @@ describe('Nav', function () {
     let instance = ReactTestUtils.renderIntoDocument(
       <Navbar componentClass={'header'}/>
     );
-    assert.ok(React.findDOMNode(instance).nodeName, 'HEADER');
+    assert.equal(React.findDOMNode(instance).nodeName, 'HEADER');
```
Comment (jquense) [inline test/NavbarSpec.js]: is this is just an unrelated fix? thanks :)

File: src/Jumbotron.js (hunk 1 above)
Comment (lo1tuma) [inline src/Jumbotron.js]: Good point. I copied this from the [`Row` component](https://github.com/react-bootstrap/react-bootstrap/blob/1bc32530e1c8cfa861ad7bbfa4bb943a3b0ecd8d/src/Row.js#L5-L7). [`Navbar`](https://github.com/react-bootstrap/react-bootstrap/blob/1bc32530e1c8cfa861ad7bbfa4bb943a3b0ecd8d/src/Navbar.js#L18) uses `node` as well.

The react documentation says the following about the `node` PropType:
> Anything that can be rendered: numbers, strings, elements or an array containing these types.

That said, I think `node` is the correct type, consider the following example:

```js
<Jumbotron componentClass={MyCustomSectionComponent}>
   <strong>foo</strong>
</Jumbotron>
```

File: test/NavbarSpec.js (hunk 2 above)
Comment (lo1tuma) [inline test/NavbarSpec.js]: Yes, I've came across this issue as I looked how other components are testing the `componentClass` property.

I’ve made it a separate commit. If desired I can make a separate PR for this change.

File: src/Jumbotron.js (hunk 1 above)
Comment (jquense) [inline src/Jumbotron.js]: A class isn't an Element in this case: https://jsfiddle.net/69z2wepo/10729/ so your example actually would fail 

also this example would not fail 
```
<Jumbotron componentClass={2}>
   <strong>foo</strong>
</Jumbotron>
```

File: test/NavbarSpec.js (hunk 2 above)
Comment (jquense) [inline test/NavbarSpec.js]: The separate commit is perfect, no need for a seperate PR, thanks for taking the time

File: src/Jumbotron.js (hunk 1 above)
Comment (lo1tuma) [inline src/Jumbotron.js]: You are correct.

What `PropType` would you suggest to use? React itself [uses `any`](https://github.com/facebook/react/blob/ba81b60ad8e93b747be42a03b797065932c49c96/src/addons/transitions/ReactTransitionGroup.js#L25).

What about the [other occurrences](https://github.com/react-bootstrap/react-bootstrap/search?utf8=%E2%9C%93&q=componentClass) of `componentClass?

Comment (jquense) [inline src/Jumbotron.js]: In the past I've personally used:
```
PropTypes.oneOf([PropTypes.string, PropTypes.func])
```
It doesn't catch everything but it gets closer at least :/ the more robust solution might be to add something to the CustomProptypes util like this: https://github.com/jquense/react-widgets/blob/master/src/util/propTypes.js#L10

for now though its probably fine to just go with `any` unless you want to tackle a more complete solution.

**Comment by mtscout6** [comment]
Since the fix to the tests is not relevant to the public api it may just be noise in the changelog so no real need for the [fixed] prefix on that commit. Also on the `[added]` commit you are repeating yourself by saying `[added] Add` you could just drop the second `Add` ;)

File: src/Jumbotron.js (hunk 1 above)
Comment (lo1tuma) [inline src/Jumbotron.js]: Ok, I’ve changed it to `any`. A more complete solution would still make sense, but I think it is out of the scope of this PR, because it should be fixed for all other components with a `componetClass` prop too.

**Comment by lo1tuma** [comment]
@mtscout6 I’ve updated the commit messages :wink:

**Comment by jquense** [comment]
LGTM

**Comment by taion** [comment]
Sorry for the minor comments but I thought we weren't going to do isRequired for props with defaults... not that we're being particularly consistent here.

Should we make an issue to fix other instances of componentClass in this repo? They all use `node`.

**Comment by mtscout6** [comment]
Yes we should make an issue to clean those up. Another issue to clean up the unnecessary `isRequired` specifiers would be good too.

**Comment by mtscout6** [comment]
LGTM
